feat(additionalcrudevent.service.ts, firebase.service.ts, eventchecki…

abhiamishra opened the pull request:
…n.service.ts): data migration

Creating three services to implement data migration from old firebase to new mongodb database

BREAKING CHANGE: Some of the services have errors in them that need to be resolved

Works on issue #98

abhiamishra committed:
feat(additionalcrudevent.service.ts, firebase.service.ts, eventcheckin.service.ts): data migration

Creating three services to implement data migration from old firebase to new mongodb database

BREAKING CHANGE: Some of the services have errors in them that need to be resolved

Works on issue #98

abhiamishra committed:
fix(additionalcrudevent.service.ts): fixed prisma error

Fixed prisma error in the query for function findOldEventID()

abhiamishra committed:
fix(fixing oldevent.resolver.ts): fixing resolvers for data migration

Fixing OldEvent.resolver.ts along with other services to make the data migration workflow work

BREAKING CHANGE: currently, oldevent.resolver.ts does not work and there is an error in
AdditionalCRUDEvent (however the code runs through so those will have to be looked at)

works on #98

abhiamishra committed:
fix(fixing function of oldevent resolver): fixed checkin for old events for data migration

Fixed string-error in AdditionalCRUD and fixed null error in OldEvent Resolver

DedsecKnight commented on apps/portal/lib/graphql/services/AdditionalCRUDEvent.service.ts:
```diff
@@ -20,4 +20,23 @@ export default class AdditionalCRUDEventService {
+    var temp = event_names[0];
+
+    for (var i = 0; i < temp.length; i++) {
+      var elem = JSON.parse(JSON.stringify(temp[i]));
```
Change `var` to `const`

DedsecKnight commented on apps/portal/lib/graphql/services/AdditionalCRUDEvent.service.ts:
```diff
@@ -20,4 +20,23 @@ export default class AdditionalCRUDEventService {
+    const prisma = getPrismaConnection();
+
+    const event_desc = [];
+    var temp = event_names[0];
```
Change `var` to `const`

DedsecKnight commented on apps/portal/lib/graphql/services/Firebase.service.ts:
```diff
@@ -0,0 +1,49 @@
+    this.db = getFirestore();
+  }
+
+  async returnEventsbyProfile(netId: string, email: string): Promise<string[]> {
```
I think we can combine line 32 and line 33 together.

DedsecKnight commented on apps/portal/lib/graphql/services/Firebase.service.ts:
```diff
@@ -0,0 +1,49 @@
+    });
+    // console.log("HEY")
+    // console.log(events)
+
```
Remove `console.log`.

DedsecKnight commented on apps/portal/lib/graphql/resolvers/OldEvent.resolver.ts:
```diff
@@ -0,0 +1,49 @@
+    @Arg('netID', () => String) netId: string,
+    @Arg('email', () => String) email: string,
+    @Ctx() context: TContext,
+  ) {
```
Add return type here.

abhiamishra replied:
Uhh what exactly do you mean? Don't we have a return type as @Mutation(() =>[EventCheckin])

DedsecKnight replied:
Yes, but the `@Mutation(() => [EventCheckin])` is type annotation for generating GraphQL schema. We also need type annotation for TypeScript function too. If you refer to other resolver functions, you will notice that each function will look something like the following: 
```js
@Mutation(() => ReturnType)
async foo(@Arg("someArg", () => SomeType) someArg: SomeType) : Promise<ReturnType> {
    // do something here
} 
```
In short, we just need the `Promise<ReturnType>` equivalent for the function in question here, which should be `Promise<EventCheckin[]>` in this case.

DedsecKnight commented on apps/portal/lib/graphql/services/Firebase.service.ts:
```diff
@@ -0,0 +1,49 @@
+
+@singleton()
+export default class Firebase {
+  db: any;
```
Can you change this to anything else other than `any`?

abhiamishra replied:
Nope - getFirestore - when hovered over - gives value of any

DedsecKnight replied:
Ok. I'll resolve this conversation then.

DedsecKnight commented on apps/portal/lib/graphql/services/Firebase.service.ts:
```diff
@@ -0,0 +1,49 @@
+import { singleton } from 'tsyringe';
+const { initializeApp, getApps, applicationDefault, cert } = require('firebase-admin/app');
+const { getFirestore, Timestamp, FieldValue } = require('firebase-admin/firestore');
+const serviceAccount = require('C:/UTD/Senior Year/Spring Semester 2022/Projects/try-acm/practice-5d375-firebase-adminsdk-tbfef-003353b527.json');
```
This is not supposed to be here.

abhiamishra replied:
Ok

DedsecKnight commented on apps/portal/lib/graphql/services/Firebase.service.ts:
```diff
@@ -0,0 +1,49 @@
+require('dotenv').config();
+
+@singleton()
+export default class Firebase {
```
potentially nit: Consider changing the name from `Firebase` to `FirebaseService`, just in case the keyword `Firebase` conflicts with something in `firebase-admin`.

abhiamishra replied:
Ok

abhiamishra committed:
refactor(oldeventresolver.ts, firebaseservice.ts, and additionalcrudevent.service.ts): refactor code

Added return type to resolver and changed variable types/filenames for refactoring purposes

DedsecKnight reviewed (approved):
LGTM 👍
